roles/yarn/tasks/main.yml

```diff
@@ -1,41 +1,48 @@
-- name: "Yarn | GPG"
-  apt_key:
-    url: https://dl.yarnpkg.com/debian/pubkey.gpg
-    state: present
-
-- name: "Yarn | Ensure Debian sources list file exists"
-  file:
-    path: /etc/apt/sources.list.d/yarn.list
-    owner: root
-    mode: '0644'
-    state: touch
-
-- name: "Yarn | Ensure Debian package is in sources list"
-  lineinfile:
-    dest: /etc/apt/sources.list.d/yarn.list
-    regexp: 'deb http://dl.yarnpkg.com/debian/ stable main'
-    line: 'deb http://dl.yarnpkg.com/debian/ stable main'
-    state: present
-
-- name: "Yarn | Update APT cache"
-  apt:
-    update_cache: yes
-
-- name: "Yarn | Install"
-  package:
-    name: yarn
-    state: latest
-  when: internet_available and is_debuntu
-
-
-# RECORD Yarn AS INSTALLED
-
-- name: "Set 'yarn_installed: True'"
-  set_fact:
-    yarn_installed: True
-
-- name: "Add 'yarn_installed: True' to {{ iiab_state_file }}"
-  lineinfile:
-    path: "{{ iiab_state_file }}"    # /etc/iiab/iiab_state.yml
-    regexp: '^yarn_installed'
-    line: 'yarn_installed: True'
+# "How do i fail a task in Ansible if the variable contains a boolean value?
+# I want to perform input validation for Ansible playbooks"
+# https://stackoverflow.com/questions/46664127/how-do-i-fail-a-task-in-ansible-if-the-variable-contains-a-boolean-value-i-want/46667499#46667499
+
+# We assume 0-init/tasks/validate_vars.yml has DEFINITELY been run, so no need
+# to re-check whether vars are defined here.  As Ansible vars cannot be unset:
+# https://serverfault.com/questions/856729/how-to-destroy-delete-unset-a-variable-value-in-ansible
+
+- name: Assert that "yarn_install is sameas true" (boolean not string etc)
+  assert:
+    that: yarn_install is sameas true
+    fail_msg: "PLEASE SET 'yarn_install: True' e.g. IN: /etc/iiab/local_vars.yml"
+    quiet: yes
+
+- name: Assert that "yarn_enabled | type_debug == 'bool'" (boolean not string etc)
+  assert:
+    that: yarn_enabled | type_debug == 'bool'
+    fail_msg: "PLEASE GIVE VARIABLE 'yarn_enabled' A PROPER (UNQUOTED) ANSIBLE BOOLEAN VALUE e.g. IN: /etc/iiab/local_vars.yml"
+    quiet: yes
+
+- debug:
+    var: yarn_install
+- debug:
+    var: yarn_enabled
+- debug:
+    var: yarn_installed
+
```

Comment:
I see debug was removed elsewhere, going to release with debug enabled?

Reply:
For now yes, intentionally displaying the 3 core variables withing dependencies for the coming months.  I think I've got these 5 covered, but might want to add a few others later?!

- MongoDB
- MySQL
- PostgreSQL
- Node.js
- Yarn

Even later this year this kind of thing might still make sense to educate new implementers.